media_driver/linux/common/os/i915/mos_bufmgr.c

```diff
@@ -4431,8 +4431,194 @@ mos_gem_bo_is_exec_object_async(struct mos_linux_bo *bo)
+    INTEL_HWCONFIG_MIN_GS_URB_ENTRIES,          /* 35 */
+    INTEL_HWCONFIG_MAX_GS_URB_ENTRIES,          /* 36 */
+    INTEL_HWCONFIG_MIN_DS_URB_ENTRIES,          /* 37 */
+    INTEL_HWCONFIG_MAX_DS_URB_ENTRIES,          /* 38 */
+    INTEL_HWCONFIG_PUSH_CONSTANT_URB_RESERVED_SIZE,     /* 39 */
+    INTEL_HWCONFIG_POCS_PUSH_CONSTANT_URB_RESERVED_SIZE,    /* 40 */
+    INTEL_HWCONFIG_URB_REGION_ALIGNMENT_SIZE_IN_BYTES,  /* 41 */
+    INTEL_HWCONFIG_URB_ALLOCATION_SIZE_UNITS_IN_BYTES,  /* 42 */
+    INTEL_HWCONFIG_MAX_URB_SIZE_CCS_IN_BYTES,       /* 43 */
+    INTEL_HWCONFIG_VS_MIN_DEREF_BLOCK_SIZE_HANDLE_COUNT,    /* 44 */
+    INTEL_HWCONFIG_DS_MIN_DEREF_BLOCK_SIZE_HANDLE_COUNT,    /* 45 */
+    INTEL_HWCONFIG_NUM_RT_STACKS_PER_DSS,           /* 46 */
+    INTEL_HWCONFIG_MAX_URB_STARTING_ADDRESS,        /* 47 */
+    INTEL_HWCONFIG_MIN_CS_URB_ENTRIES,          /* 48 */
+    INTEL_HWCONFIG_MAX_CS_URB_ENTRIES,          /* 49 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_URB,           /* 50 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_REST,          /* 51 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_DC,            /* 52 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_RO,            /* 53 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_Z,         /* 54 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_COLOR,         /* 55 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_UNIFIED_TILE_CACHE,    /* 56 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_COMMAND_BUFFER,    /* 57 */
+    INTEL_HWCONFIG_L3_ALLOC_PER_BANK_RW,            /* 58 */
+    INTEL_HWCONFIG_MAX_NUM_L3_CONFIGS,          /* 59 */
+    INTEL_HWCONFIG_BINDLESS_SURFACE_OFFSET_BIT_COUNT,   /* 60 */
+    INTEL_HWCONFIG_RESERVED_CCS_WAYS,           /* 61 */
+    INTEL_HWCONFIG_CSR_SIZE_IN_MB,              /* 62 */
+    INTEL_HWCONFIG_GEOMETRY_PIPES_PER_SLICE,        /* 63 */
+    INTEL_HWCONFIG_L3_BANK_SIZE_IN_KB,          /* 64 */
+    INTEL_HWCONFIG_SLM_SIZE_PER_DSS,            /* 65 */
+    INTEL_HWCONFIG_MAX_PIXEL_FILL_RATE_PER_SLICE,       /* 66 */
+    INTEL_HWCONFIG_MAX_PIXEL_FILL_RATE_PER_DSS,     /* 67 */
+    INTEL_HWCONFIG_URB_SIZE_PER_SLICE_IN_KB,        /* 68 */
+    INTEL_HWCONFIG_URB_SIZE_PER_L3_BANK_COUNT_IN_KB,    /* 69 */
+    INTEL_HWCONFIG_MAX_SUBSLICE,                /* 70 */
+    INTEL_HWCONFIG_MAX_EU_PER_SUBSLICE,         /* 71 */
+    INTEL_HWCONFIG_RAMBO_L3_BANK_SIZE_IN_KB,        /* 72 */
+    INTEL_HWCONFIG_SLM_SIZE_PER_SS_IN_KB,           /* 73 */
+    __INTEL_HWCONFIG_LIMIT
+};
+
+enum {
+    INTEL_HWCONFIG_MEMORY_TYPE_LPDDR4 = 0,
+    INTEL_HWCONFIG_MEMORY_TYPE_LPDDR5,
+};
+
+#define BIT(x) (1ul <<(x))
+enum {
+    INTEL_HWCONFIG_CACHE_TYPE_L3    = BIT(0),
+    INTEL_HWCONFIG_CACHE_TYPE_LLC   = BIT(1),
+    INTEL_HWCONFIG_CACHE_TYPE_EDRAM = BIT(2),
+};
+
+#ifndef DRM_I915_QUERY_HWCONFIG_TABLE
+#define DRM_I915_QUERY_HWCONFIG_TABLE   5
+#endif
+
+static int mos_query_items(int fd, struct drm_i915_query_item *query_item)
+{
+    struct drm_i915_query query;
+    int ret;
+
+    assert(query_item);
```

Comment:
could we use assert? if release or release internal version, usually crash instead of assert. Or i am wrong? thank you! Just for my education purpose.

Reply:
Not sure if I understood your point. We are just following the convention of mos_bufmgr functions to use assert here to make sure query_item is a valid ptr.